pxtextension.ts

```diff
@@ -35,11 +37,13 @@ class MlEvent {
   //% blockId=mlrunner_on_ml_event
   //% block="on ML event $this"
   onEvent(body: () => void): void {
+    this.hasOnEventHandler = true;
     const wrappedBody = () => {
-      if (!mlrunner.prevAction || mlrunner.prevAction !== this) {
+      const prevAction = mlrunner.Action.prevAction;
+      mlrunner.Action.prevAction = this.eventValue;
+      if (!prevAction || prevAction !== this.eventValue) {
```

Comment:
With this implementation here we can remove the C++ track of `lastPredictionEventId`?

Reply:
I haven't thought this all the way through, but I think if the user doesn't use the `onEvent` blocks, just the `isEvent` blocks, there's no way the TypeScript layer can know what the last prediction was? The code highlighted would never run in that case.

Reply:
Ah, yes, you are right 👍